crates/forge_loader/src/manifest.rs

```diff
@@ -208,65 +434,316 @@ impl<T> AsRef<T> for FunctionTy<T> {
+            // let Some(func) = access.on_delete_import;
+            // invokable_functions.extend(func.function);
+            self.clone()
+                .add_optional(access.on_delete_import, &mut invokable_functions);
+
+            invokable_functions.extend(access.stop_import.function);
+            invokable_functions.extend(access.start_import.function);
+            invokable_functions.extend(access.import_status.function);
+        });
+        org_panel
+            .iter()
+            .for_each(|panel| panel.append_functions(&mut invokable_functions));
+
+        portal_footer
+            .iter()
+            .for_each(|footer| footer.append_functions(&mut invokable_functions));
+
+        portal_header
+            .iter()
+            .for_each(|header| header.append_functions(&mut invokable_functions));
+
+        portal_profile_panel
+            .iter()
+            .for_each(|profile| profile.append_functions(&mut invokable_functions));
+
+        portal_req
+            .iter()
+            .for_each(|req| req.append_functions(&mut invokable_functions));
+
+        portal_request_detail
+            .iter()
+            .for_each(|req| req.append_functions(&mut invokable_functions));
+
+        portal_request_detail_panel
+            .iter()
+            .for_each(|req| req.append_functions(&mut invokable_functions));
+
+        portal_request_view_action
+            .iter()
+            .for_each(|req| req.append_functions(&mut invokable_functions));
+
+        portal_subheader
+            .iter()
+            .for_each(|subheader| subheader.append_functions(&mut invokable_functions));
+        portal_header_menu_action
+            .iter()
+            .for_each(|action| action.append_functions(&mut invokable_functions));
+
+        queue_page
+            .iter()
+            .for_each(|page| page.append_functions(&mut invokable_functions));
+
+        functions.into_iter().flat_map(move |func| {
+            let web_trigger = webtriggers
+                .binary_search_by_key(&func.key, |trigger| &trigger.function)
+                .is_ok();
+            let invokable = invokable_functions.contains(func.key);
+            // this checks whether the funton being scanned is being used in an admin module. Rn it only checks for jira_admin page module.
+            // optionally: compass:adminPage could also be considered.
+            let admin = jira_admin_page.iter().any(|admin_function| {
+                let Some(string) = admin_function.common_keys.function else {
+                    return false;
+                };
+                return string == func.key;
+            }) || compass_admin_page.iter().any(|admin_function| {
+                let Some(string) = admin_function.function else {
+                    return false;
+                };
+                return string == func.key;
+            });
+
+            Ok::<_, Error>(Entrypoint {
+                function: FunctionRef::try_from(func)?,
```

Comment:
After destructuring and making all attributes refs unless necessary, Rust is throwing a trait bound error. And I need help to understand what's up. 

`error[E0277]: the trait bound `FunctionRef<'_>: From<&FunctionMod<'_>>` is not satisfied
   --> crates/forge_loader/src/manifest.rs:735:27
    |
735 |                 function: FunctionRef::try_from(func)?,
    |                           ^^^^^^^^^^^^^^^^^^^^^ the trait `From<&FunctionMod<'_>>` is not implemented for `FunctionRef<'_>`
    |
    = help: the trait `TryFrom<FunctionMod<'a>>` is implemented for `&'a FunctionRef<'a>`
    = note: required for `&FunctionMod<'_>` to implement `Into<FunctionRef<'_>>`
    = note: required for `FunctionRef<'_>` to implement `TryFrom<&FunctionMod<'_>>`
`

I'm kinda confused why it's mad that certain traits aren't implemented. But I'm guessing it has to do with how everything is a reference now so that values can be reused or smth. Anyways, this is the last blocker rn.

Reply:
The `TryFrom` impl is backwards, you want to `impl<'a> TryFrom<&FunctionMod<'a>> for FunctionRef<'a>`. The `impl <'a> TryFrom<FunctionMod<'a>> for &'a FunctionRef<'a>` can only go from owned `FunctionMod<'a>` to borrowed `FunctionRef<'a>`.

To fix the failing test on line 885, just do `FunctionRef::try_from(&func_handler).unwrap()` instead of using `try_into`.